include/libtorrent/hasher.hpp

```diff
@@ -119,6 +122,8 @@ TORRENT_CRYPTO_NAMESPACE
 		gcry_md_hd_t m_context;
 #elif TORRENT_USE_COMMONCRYPTO
 		CC_SHA1_CTX m_context;
+#elif TORRENT_USE_CNG
+		aux::cng_hash<aux::cng_algorithm::sha1> m_context;
```

Comment:
it seems like perhaps `aux::cng_hash` should be given an API compatible to `hasher`, and it could just be an alias here, instead of interleaving the cng code with the rest of `hasher`. In fact, given how many `hasher` implementations there are, that would probably be better for all of them

Reply:
just used same approach as TORRENT_USE_CRYPTOAPI.

Reply:
yeah, it's fine. I think as this grows in number of alternatives, it would make sense to split them out into individual classes (with compatible interfaces). but that's separate from this patch